### ISSUE TITLE
rmdir recursively while empty?

### ISSUE THREAD

**Issue by stuart-little** [labster/p6-file-directory-tree#17]
This is a suggestion for a possible feature (I found myself having to hand-roll the following recently, and figured it'd make for a good enhancement for `File::Directory::Tree`).

Suppose you want to remove an empty directory (like `rmdir` does), and then have that removal bubble up until you encounter a non-empty directory. So:

- for the path `/a/b/c/d`, if none of those directories have anything else inside, then running this command would remove all of them;
- for the same path, if `a/b` *also* contains something else apart from `c`, then running this command on `/a/b/c/d` would remove `c` but stop there (because afterwards `/a/b` wouldn't be empty). 
In short, something like [rmtree](https://github.com/labster/p6-file-directory-tree/blob/12f1ab52191efa38787de26e5a8d4c01a94d03a8/lib/File/Directory/Tree.pm#L35) but only calling `rmdir` and not `empty-directory`. 

Perhaps whether `rmtree` calls `empty-directory` or not can be passed as a named parameter.
